invenio_cli/helpers/scripts.py

```diff
@@ -0,0 +1,298 @@
+
+"""Previous script files helper methods."""
+
+import errno
+import logging
+import os
+import shutil
+import signal
+import subprocess
+import time
+from pathlib import Path
+
+import click
+from flask import current_app
+from flask.cli import with_appcontext
+from invenio_base.app import create_cli
+
+from .log import LogPipe
+
+
+def run_command(cli, runner, command, message=None, catch_exceptions=False,
+                verbose=False):
+    """Run [invenio] command."""
+    click.secho("{}...".format(message if message else command),
+                fg="green")
+    res = runner.invoke(cli, command, catch_exceptions=catch_exceptions)
+    if verbose:
+        click.secho(res.output)
+
+
+#############
+# BOOTSTRAP #
+#############
+
+
+def _bootstrap_dev(pre, loglevel, logfile, verbose):
+    # Check if the dependencies have been locked, fail if not
+    click.secho('Checking that dependencies are locked...', fg='green')
+    locked = 'Pipfile.lock' in os.listdir('.')
+    if not locked:
+        click.secho('Cannot build without locked dependencies', fg='red')
+        exit(1)
+
+    # Open logging pipe
+    logpipe = LogPipe(loglevel, logfile)
+
+    # Install packages according to the lock file
+    click.secho('Installing dependencies...', fg='green')
+    command = ['pipenv', 'sync', '--dev']
+
+    if pre:
+        command.append('--pre')
+    subprocess.call(command, stdout=logpipe, stderr=logpipe)
+
+    # Close logging pipe
+    logpipe.close()
+
+    # Build assets
+    build_assets(verbose)
+
+    # Update static files
+    update_statics(True, loglevel, logfile)
+
+    # Update configuration
+    update_config(True, loglevel, logfile)
+
+
+def _boostrap_prod(base, app, docker_helper, app_name):
+    if base:
+        click.secho('Building base docker image...', fg='green')
+        # docker build -f Dockerfile.base -t my-site-base:latest .
+        docker_helper.built_image(
+            dockerfile='Dockerfile.base',
+            tag='{project_name}-base:latest'.format(
+                project_name=app_name)
+        )
+    if app:
+        click.secho('Building applications docker image...', fg='green')
+        # docker build -t my-site:latest .
+        docker_helper.built_image(
+            dockerfile='Dockerfile',
+            tag='{project_name}:latest'.format(
+                project_name=app_name)
+        )
+
+
+def bootstrap(dev=True, pre=True,
+              base=True, app=True, docker_helper=None, app_name='invenio-rdm',
+              verbose=False, loglevel=logging.WARN, logfile='invenio-cli.log'):
+    """Bootstrap server."""
+    click.secho('Bootstrapping server...', fg='green')
+    if dev:
+        _bootstrap_dev(pre, loglevel, logfile, verbose)
+    else:
+        _boostrap_prod(base, app, docker_helper, app_name)
+
+
+@with_appcontext
+def build_assets(verbose):
+    """Build assets."""
+    cli = create_cli()
+    runner = current_app.test_cli_runner()
```

Comment:
Is there another runner than the test_cli one? I fear what runner's settings may be implied.

Reply:
https://github.com/inveniosoftware/invenio-cli/issues/49